src/lib.rs

```diff
@@ -285,7 +324,15 @@ impl<K: Hash + Eq, V, S: HashState> LinkedHashMap<K, V, S> {
         removed.map(|mut node| {
             let node_ptr: *mut LinkedHashMapEntry<K,V> = &mut *node;
             self.detach(node_ptr);
-            node.value
+            unsafe {
+                // add to free list
+                (*node_ptr).next = self.free;
+                self.free = node_ptr;
+                // forget the box but drop the key and return the value
+                mem::forget(node);
+                drop(ptr::read(&(*node_ptr).key));
```

Comment:
We are recycling the Box (presumably "empty") and returning the value value. So we should drop the key here to avoid memory leaks.